examples/widgets/ellipse_selector_demo.py

```diff
@@ -0,0 +1,146 @@
+"""
+
+=====================
+Ellipse Selector Demo
+=====================
+
+Interactively selecting data in a region of interest in an image by drawing an
+ellipse.
+
+This example plots an image. You can then draw an ellipse shaped selection. To
+draw, click and drag to an extent within the image and then release.
+"""
+
+import numpy as np
+
+from matplotlib.patches import Ellipse
+from matplotlib.widgets import EllipseSelector
+
+
+class EllipseROIStats(object):
+    """
+    Select a region of an image using 'EllipseSelector' and extract the data.
+    The extents and center of the ellipse drawn are used to create a patch
+    in the same area as the selector object. The patch is the then used to
+    generate a mask that can be used to extract data.
+
+    Parameters
+    ----------
+    ax: `~matplotlib.axes.Axes`: Axes to interact with
+    img_data: `numpy.ndarray`: Image array
+    """
+
+    def __init__(self, ax, img_data):
+        self.ax = ax
+        self.fig = ax.figure
+
+        lineprop = dict(color='black', linestyle="-.", linewidth=2,
+                        dashes=(3, 1), alpha=1.0)
+        state_mods = dict(move=' ', clear='escape')
+        self.ellipse_roi = EllipseSelector(self.ax, onselect=self.on_select,
+                                           drawtype='line', lineprops=lineprop,
+                                           state_modifier_keys=state_mods,
+                                           interactive=True)
+        self.img = img_data
+
+    def on_select(self, eclick, erelease):
+        """eclick and erelease are matplotlib events at press and release."""
+        # Remove previously drawn ellipse
+        self.ax.lines.clear()
+        self.ax.patches.clear()
+
+        # Starting and ending column indicies
+        self.x_start = eclick.xdata
+        self.x_end = erelease.xdata
+
+        # Starting and ending row indicies
+        self.y_start = eclick.ydata
+        self.y_end = erelease.ydata
+
+        # Have the ellipse be drawn
+        extents = [self.x_start, self.x_end, self.y_start, self.y_end]
+        self.ellipse_roi.draw_shape(extents)
+
+        # Parameters of the ellipse drawn
+        width = self.x_end - self.x_start
+        height = self.y_end - self.y_start
+        center = self.ellipse_roi.center
+
+        # Instantiate a Patch object that matches
+        # that of the drawn ellipse.
+        ellipse = Ellipse(xy=center, width=width, height=height)
```

Comment:
What do you need this ellipse for? Is it intended to stay persistent and independent of the selector? If not, you could use the selector itself.

Reply:
I am using the elliptical patch to extract intensity values from in the image that match where the ellipse is drawn.